docs/manpage.rst

```diff
@@ -379,6 +379,22 @@ Options controlling ReFrame execution
 
    .. versionadded:: 3.2
 
+.. option:: --distribute=STATE
+
+   Parameterize and run tests on all the nodes of their partitions.
+   The attribute ``pin_nodes`` will be set in these tests in order to be submitted on the selected nodes.
```

Comment:
I don't think that this is necessary here. It is more important how the mechanism works, i.e., it parametrises the selected test with the selected nodes on each of their partitions and that it uses the `$nid` parameter. Think of what the user would like to read when trying this option. The user will see a different name of her test and a strange parameter. So you need to explain those. The `pin_nodes` is an implementation detail.